sdk/core/azure-core/src/main/java/com/azure/core/http/rest/Page.java

```diff
@@ -3,26 +3,28 @@
 
 package com.azure.core.http.rest;
 
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.paging.ContinuablePage;
+
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * Represents a paginated REST response from the service.
  *
- * @param <T> Type of the listed objects in that response.
+ * @param <T> Type of items in the page response.
  */
-public interface Page<T> {
-
-    /**
-     * Gets a list of items returned from the service.
-     *
-     * @return A list of items from the service.
-     */
-    List<T> getItems();
-
+public interface Page<T> extends ContinuablePage<String, T> {
```

Comment:
Two things here:
 
1. We need a base type to represent the general page contract, we don’t want that contract to be in `http` package since non-http based client libs might want to use it. A genuine concern from cosmos team was  - inheritance hierarchy of TCP based cosmos type containing core types from `http` package, the `Page` type is in `http` package.
2. Ideally we could merge `Page` and `ContinuablePage` and call it `Page` then move it out of `http` package but that is a breaking change.
 
So we’ve above two constraints - hence the type ContinuablePage and inheritance hierarchy as proposed in this PR.